lib/service/chat_service.dart

```diff
@@ -0,0 +1,223 @@
+import 'dart:convert';
+import 'dart:io';
+import 'dart:typed_data';
+
+import 'package:autonomy_flutter/common/environment.dart';
+import 'package:autonomy_flutter/common/injector.dart';
+import 'package:autonomy_flutter/model/chat_message.dart' as app;
+import 'package:autonomy_flutter/model/pair.dart';
+import 'package:autonomy_flutter/service/chat_auth_service.dart';
+import 'package:autonomy_flutter/service/tezos_service.dart';
+import 'package:autonomy_flutter/util/log.dart';
+import 'package:crypto/crypto.dart';
+import 'package:eth_sig_util/util/utils.dart';
+import 'package:libauk_dart/libauk_dart.dart';
+import 'package:web_socket_channel/io.dart';
+import 'package:web_socket_channel/web_socket_channel.dart';
+
+abstract class ChatService {
+  static const ERROR = "error";
+  static const SENT = "sent";
+
+  Future<void> connect({
+    required String address,
+    required String id,
+    required Pair<WalletStorage, int> wallet,
+  });
+
+  void addListener(ChatListener listener);
+
+  void removeListener();
+
+  void sendMessage(dynamic message);
+
+  Future<void> dispose();
+}
+
+class PostcardChatService implements ChatService {
```

Comment:
looks like there is no specific logic for postcards in this service. I think we can simply rename to  `ChatServiceImpl`.